components/lcm/org.wso2.carbon.apimgt.lifecycle.manager.core/src/main/java/org/wso2/carbon/apimgt/lifecycle/manager/core/internal/LCServiceComponent.java

```diff
@@ -48,70 +49,23 @@ protected void activate(ComponentContext context) {
             log.debug("Social Activity service is activated  with SQL Implementation");
         }
 
-        JDBCPersistenceManager jdbcPersistenceManager;
         try {
-            jdbcPersistenceManager = JDBCPersistenceManager.getInstance();
-            jdbcPersistenceManager.initializeDatabase();
-        } catch (Exception e) {
-            log.error("Failed to initilize database. " + e);
-        }
-        testAddinLC();
-        testGettingLCList();
-        testGetLCContent();
-    }
-
-    protected void setDataSourceService(DataSourceService dataSourceService) {
-        if (log.isDebugEnabled()) {
-            log.debug("Setting the DataSourceService");
-        }
-        JDBCPersistenceManager.setCarbonDataSourceService(dataSourceService);
-    }
+            LCMgtDBUtil.initialize();
+            LCUtils.initiateLCMap();
 
-    protected void unsetDataSourceService(DataSourceService dataSourceService) {
-        if (log.isDebugEnabled()) {
-            log.debug("Unsetting the DataSourceService");
-        }
-        JDBCPersistenceManager.setCarbonDataSourceService(null);
-    }
-
-    private void testAddinLC(){
-        LCManagementService lcManagementService = new LCManagementServiceImpl();
-        String content =
-                "<aspect name=\"ServerLifecycle\" class=\"org.wso2.carbon.governance.registry.extensions.aspects.DefaultLifeCycle\">\n"
-                + "    <configuration type=\"literal\">\n" + "        <lifecycle>\n"
-                + "            <scxml xmlns=\"http://www.w3.org/2005/07/scxml\"\n"
-                + "                   version=\"1.0\"\n" + "                   initialstate=\"Active\">\n"
-                + "                <state id=\"Active\">\n"
-                + "                    <transition event=\"Deactivate\" target=\"Off\"/>\n"
-                + "                </state>\n" + "                <state id=\"Off\">\n"
-                + "                    <transition event=\"Activate\" target=\"Active\"/>\n"
-                + "                </state>\n" + "            </scxml>\n" + "        </lifecycle>\n"
-                + "    </configuration>\n" + "</aspect>";
-        try {
-            lcManagementService.createLifecycle(content);
+        } catch (LCManagerDatabaseException e) {
+            log.error("Failed to initialize database. " + e);
```

Comment:
shall we pass the exception like this? 

log.error("Failed to initialize database. " + e.getMessage() , e);